Action fails when used with a container job

Given the following workflow:

```yaml
name: Cache Test

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  publish:
    runs-on: ubuntu-latest
    container: mcr.microsoft.com/powershell:ubuntu-22.04
    steps:

      - uses: actions/checkout@v3

      - name: PowerShell Module Cache
        uses: potatoqualitee/psmodulecache@v5.2
        with:
          modules-to-cache: Az.Network
```

The action fails:
![image](https://user-images.githubusercontent.com/384747/210121182-99034bda-c4d6-4cb1-9d19-8d7acdc32a69.png)

Here's the full output from the action:

```text
Run potatoqualitee/psmodulecache@v5.2
  with:
    modules-to-cache: Az.Network
    shell: pwsh
    updatable: false
    prefixidentifier: false
Run Import-Module /home/runner/work/_actions/potatoqualitee/psmodulecache/v5.2/PSModuleCache.psd1
  Import-Module /home/runner/work/_actions/potatoqualitee/psmodulecache/v5.2/PSModuleCache.psd1
  Write-Output "Getting variables for cacher"
  
  $Parameters = @{
    Modules = "Az.Network"
    PrereleaseModules = ""
    Shells = "pwsh"
    Updatable = [boolean]::Parse("false")
    PrefixIdentifier = [boolean]::Parse("false")
  }
  Write-Output "Action Parameters = $([PsCustomObject]$Parameters)"
  
  $ActionParameters = New-ModuleCacheParameter @Parameters
  
  $ModuleCaches = Get-ModuleCache $ActionParameters
  "keygen=$($ModuleCaches.Key)" >> $env:GITHUB_OUTPUT
  
  $Paths = @(New-ModuleSavePath $ModuleCaches.ModuleCacheInformations)
  
  # Accommodate multi-line
  "modulepath<<EOF" >> $env:GITHUB_OUTPUT
  $Paths >> $env:GITHUB_OUTPUT
  "EOF" >> $env:GITHUB_OUTPUT
  
  $null = $ModuleCaches | Export-CliXml -Path (Join-Path $home -ChildPath $CacheFileName)
  shell: pwsh -command ". '{0}'"
  
Import-Module: /__w/_temp/393f6afa-82ee-45c2-8362-0fdf70641abd.ps1:2
Line |
   2 |  Import-Module /home/runner/work/_actions/potatoqualitee/psmodulecache …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The specified module
     | '/home/runner/work/_actions/potatoqualitee/psmodulecache/v5.2/PSModuleCache.psd1' was not loaded because no valid module file was found in any module directory.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/Container.yml
+++ b/.github/workflows/Container.yml
@@ -1,0 +1,23 @@
+name: Cache Test
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  container:
+    runs-on: ubuntu-latest
+    container: mcr.microsoft.com/powershell:ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          Get-ChildItem Env: | Sort-Object Name | Format-Table -Wrap -AutoSize
+        shell: pwsh
+
+      - name: Cache modules
+        id: psmodulecache
+        uses: ./
+        with:
+          modules-to-cache: Az.Network
+          shell: pwsh
+
+      - run: Get-InstalledModule Az.Network
+        shell: pwsh

--- a/.github/workflows/Container.yml
+++ b/.github/workflows/Container.yml
@@ -16,8 +16,15 @@ jobs:
         id: psmodulecache
         uses: ./
         with:
-          modules-to-cache: Az.Network
+          modules-to-cache: powershell-yaml
           shell: pwsh
 
-      - run: Get-InstalledModule Az.Network
+      - run: |
+          Get-InstalledModule powershell-yaml
+
+          Import-Module powershell-yaml
+
+          $yaml = (Get-Content action.yml) | ConvertFrom-Yaml
+
+          $yaml
         shell: pwsh

--- a/PSModuleCache.psm1
+++ b/PSModuleCache.psm1
@@ -463,7 +463,10 @@ function Get-ModuleSavePath {
       }
 
    } elseif ($env:RUNNER_OS -in @('Linux', 'MacOS')) {
-      $null = sudo chown -R runner $script:PsLinuxCoreModulePath
+      # Ensure 'sudo' exists
+      if (get-command sudo -ErrorAction Ignore) {
+         $null = sudo chown -R runner $script:PsLinuxCoreModulePath
+      }
       Write-Output $script:PsLinuxCoreModulePath
    } else {
       throw "`$env:RUNNER_OS ('$env:RUNNER_OS') is empty or unknown."

--- a/PSModuleCache.psm1
+++ b/PSModuleCache.psm1
@@ -306,6 +306,7 @@ function New-ModuleCache {
    function Split-ModuleParameter {
       #return one or more 'ModuleToCacheInformation' objects.
       param(
+         [bool]$ContainerJob,
          $modules,
          $ShellsParameter,
          [switch]$AllowPrerelease
@@ -316,7 +317,7 @@ function New-ModuleCache {
 
          if ($null -ne $cacheinfo) {
             #To save a module, we use the PSModulePath associated with the shells,the module name and the version (not a semver).
-            $cacheinfo.ModuleSavePaths = Get-ModuleSavePath -Shells $ShellsParameter
+            $cacheinfo.ModuleSavePaths = Get-ModuleSavePath -ContainerJob $ContainerJob -Shells $ShellsParameter
          }
          Write-Output $cacheinfo
       }
@@ -390,8 +391,8 @@ function New-ModuleCache {
    $PrereleaseModules = $Action.PrereleaseModulesParameter.ToArray()
 
    $ModuleCacheInformations = @(
-      Split-ModuleParameter -modules $StableModules -ShellsParameter $Action.ShellsParameter
-      Split-ModuleParameter -modules $PrereleaseModules -ShellsParameter $Action.ShellsParameter -AllowPrerelease
+      Split-ModuleParameter -ContainerJob $Action.ContainerJob -modules $StableModules -ShellsParameter $Action.ShellsParameter
+      Split-ModuleParameter -ContainerJob $Action.ContainerJob -modules $PrereleaseModules -ShellsParameter $Action.ShellsParameter -AllowPrerelease
    )
 
    #The serialization depth use default value 2
@@ -449,6 +450,7 @@ function Get-ModuleSavePath {
    It is assumed that the parameters are valid and have been tested before calling this function.
 #>
    param(
+      [bool]$ContainerJob,
       #Depending on the implicit rule of the 'Shell' parameter of the Action, contains either 'powershell' or 'pwsh' or both.
       [string[]]$Shells
    )
@@ -464,7 +466,7 @@ function Get-ModuleSavePath {
 
    } elseif ($env:RUNNER_OS -in @('Linux', 'MacOS')) {
       # Ensure 'sudo' exists
-      if (get-command sudo -ErrorAction Ignore) {
+      if (-not $ContainerJob) {
          $null = sudo chown -R runner $script:PsLinuxCoreModulePath
       }
       Write-Output $script:PsLinuxCoreModulePath
@@ -616,7 +618,8 @@ function New-ModuleCacheParameter {
       [string]$Shells,
 
       [switch]$Updatable,
-      [switch]$PrefixIdentifier
+      [switch]$PrefixIdentifier,
+      [bool]$ContainerJob
    )
 
    if (($Modules.Trim() -eq [string]::Empty) -and ($PrereleaseModules.Trim() -eq [string]::Empty) )
@@ -642,6 +645,7 @@ function New-ModuleCacheParameter {
       CacheType                  = [CacheType]$(if ($Updatable.IsPresent) { [CacheType]::Updatable } else { [CacheType]::Immutable } )
       Updatable                  = $Updatable;
       PrefixIdentifier           = $PrefixIdentifier
+      ContainerJob               = $ContainerJob
    }
 
    $sbToArray = {

--- a/action.yml
+++ b/action.yml
@@ -29,7 +29,7 @@ runs:
     - id: psoutput
       shell: pwsh
       run: |
-        Import-Module ${{ github.action_path }}/PSModuleCache.psd1
+        Import-Module $env:GITHUB_ACTION_PATH/PSModuleCache.psd1
         Write-Output "Getting variables for cacher"
 
         $Parameters = @{
@@ -70,5 +70,5 @@ runs:
       run: |
         Write-Output "Saving all modules"
 
-        Import-Module ${{ github.action_path }}/PSModuleCache.psd1
+        Import-Module $env:GITHUB_ACTION_PATH/PSModuleCache.psd1
         Save-ModuleCache

--- a/action.yml
+++ b/action.yml
@@ -38,6 +38,7 @@ runs:
           Shells = "${{ inputs.shell }}"
           Updatable = [boolean]::Parse("${{ inputs.updatable }}")
           PrefixIdentifier = [boolean]::Parse("${{ inputs.prefixidentifier }}")
+          ContainerJob = ("${{ job.container.id}}" -ne "")
         }
         Write-Output "Action Parameters = $([PsCustomObject]$Parameters)"
 

